web/server/codechecker_server/api/report_server.py

```diff
@@ -1562,22 +1562,38 @@ def _setReviewStatus(self, report_id, status, message, session):
                 review_status = ReviewStatus()
                 review_status.bug_hash = report.bug_id
 
-            user = self.__get_username()
-
-            old_status = review_status.status if review_status.status \
-                else review_status_str(ttypes.ReviewStatus.UNREVIEWED)
-            old_msg = review_status.message.decode('utf-8') \
-                if review_status.message else None
-
-            review_status.status = review_status_str(status)
-            review_status.author = user
-            review_status.message = message.encode('utf8') if message else b''
-            review_status.date = datetime.now()
-            session.add(review_status)
+            old_status = review_status.status or \
+                review_status_str(ttypes.ReviewStatus.UNREVIEWED)
+            old_msg = review_status.message or None
+
+            new_status = review_status_str(status)
+            new_user = self.__get_username()
+            new_message = message.encode('utf8') if message else b''
+
+            # Review status is a shared table among runs. When multiple runs
+            # are stored in parallel, there may be a race condition in updating
+            # review status fields. The most common reason of deadlocks is
+            # changing only the date to current date. This condition checks if
+            # something else is also changed other than dates.
+            # TODO: This condition may not be necessary if mass storage is
+            # redesigned so storing transactions run sequentially.
+            if (old_status, review_status.author, old_msg) == \
+                    (new_status, new_user, new_message):
+                return True
 
-            # Create a system comment if the review status or the message is
-            # changed.
-            if old_status != review_status.status or old_msg != message:
+            if old_status != new_status or old_msg != new_message:
+                # We assume that report status in source code comments belong
+                # to the first user who stored the reports. If another user
+                # stores the same project with same report status then we don't
+                # change it.
+                review_status.status = new_status
+                review_status.author = new_user
+                review_status.message = new_message
+                review_status.date = date or datetime.now()
```

Comment:
Could you please use this `date` variable when creating a system comment below? This way the system comment will have the same date as the storage date too.